uaclient/api/u/pro/status/is_attached/v1.py

```diff
@@ -0,0 +1,29 @@
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.config import UAConfig
+from uaclient.data_types import BoolDataValue, DataObject, Field
+
+
+class IsAttachedResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("is_attached", BoolDataValue),
+    ]
+
+    def __init__(self, *, is_attached: bool):
+        self.is_attached = is_attached
+
+
+def is_attached() -> IsAttachedResult:
+    return _is_attached(UAConfig())
+
+
+def _is_attached(cfg: UAConfig) -> IsAttachedResult:
+    return IsAttachedResult(is_attached=cfg.is_attached)
```

Comment:
Maybe this is too much for this PR, but this is also an opportunity to decouple `is_attached` from UAConfig.

We could directly check for the existence of the machine token file here and remove the property from UAConfig entirely - thoughts?

Reply:
Good idea, I think that is doable